src/eva/core/callbacks/writers/embeddings/base.py

```diff
@@ -163,6 +166,21 @@ def _get_item_metadata(
 
         return item_metadata
 
+    def _check_if_exists(self) -> None:
+        """Checks if the output directory already exists and if it should be overwritten."""
+        exists = (
+            os.path.isdir(self._output_dir) and len(list(Path(self._output_dir).rglob("*.pt"))) > 0
+        )
+        if exists and not self._overwrite:
+            raise FileExistsError(
+                f"Embeddings already exists in {self._output_dir}. This either means that they "
+                "have been computed before or that a wrong output directory is being used. "
+                "Consider using `eva fit` instead, selecting a different output directory or "
+                "setting overwrite=True."
+            )
```

Comment:
this could be an alternative, but not sure if a good idea
```py3
try:
   os.makedirs(self._output_dir, exist_ok=self._overwrite)
except FileExistsError as e:
            raise FileExistsError(
                f"Embeddings already exists in {self._output_dir}. This either means that they "
                "have been computed before or that a wrong output directory is being used. "
                "Consider using `eva fit` instead, selecting a different output directory or "
                "setting overwrite=True."
            )

Reply:
done